subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/StrictDependenciesIntegrationTest.groovy

```diff
@@ -102,15 +104,28 @@ class StrictDependenciesIntegrationTest extends AbstractHttpDependencyResolution
 
     void "should pass if transitive dependency version matches exactly the strict dependency version"() {
         given:
-        def foo10 = mavenHttpRepo.module("org", "foo", '1.0').publish()
-        def bar10 = mavenHttpRepo.module('org', 'bar', '1.0')
-            .dependsOn(foo10)
-            .publish()
+        repository {
+            group('org') {
+                module('foo') {
+                    version('1.0') {
+                        expectGetMetadata()
```

Comment:
I don't think that having expectations here is better. It doesn't feel like test setup, and makes the setup less readable. Instead, I'd like to see these specific interactions listed in the `then:` clause.

Something like:
```
then:
repository.module('org:foo:1.0') {
    expectGetMetadata()
    expectGetArtifact()
}
```

This is separate from the discussion of which tests should actually verify interactions at all.

Reply:
Yes, I would also prefer to see them in a `then` block. But then (no pun intended) it requires a different backing HTTP test fixture, because the current ones setup requirements.

Reply:
Well we _could_ leave them in the `when:` block. As it currently stands, the test author doesn't have the option to put expectations elsewhere, correct? (Because they don't have a reference to the produced module).